### ISSUE TITLE
Prevent duplicate selector roots

### ISSUE THREAD

**Issue by nschonni** [brigade/scss-lint#109]
Simple case
**Bad**

``` scss
.foo {
  color: red;
}

//More rules

.foo {
  .bar {
    color: blue;
  }
}
```

**Good**

``` scss
.foo {
  color: red;
  .bar {
    color: blue;
  }
}
```

This may be more complicated when looking at nested selectors
